test/Dialect/Calyx/errors.mlir

```diff
@@ -125,3 +125,81 @@ calyx.program {
     calyx.control {}
   }
 }
+
+// -----
+
+calyx.program {
+  calyx.component @A(%go: i1, %clk: i1, %reset: i1) -> (%out: i8, %done: i1) {
+    calyx.wires {}
+    calyx.control {}
+  }
+  calyx.component @B(%in: i32, %go: i1, %clk: i1, %reset: i1) -> (%out: i32, %done: i1) {
+    %c0.go, %c0.clk, %c0.reset, %c0.out, %c0.done = calyx.cell "c0" @A : i1, i1, i1, i8, i1
+    calyx.wires { calyx.group @Group1 {} }
+    calyx.control {
+      calyx.seq {
+        // expected-error @+2 {{use of value '%c0.out' expects different type than prior uses: 'i1' vs 'i8'}}
+        // expected-note @-5 {{prior use here}}
+        calyx.if %c0.out, @Group1 { } else { }
+      }
+    }
+  }
+}
+
+// -----
+
+calyx.program {
+  calyx.component @A(%go: i1, %clk: i1, %reset: i1) -> (%out: i1, %done: i1) {
+    calyx.wires {}
+    calyx.control {}
+  }
+  calyx.component @main(%in: i32, %go: i1, %clk: i1, %reset: i1) -> (%out: i32, %done: i1) {
+    %c0.go, %c0.clk, %c0.reset, %c0.out, %c0.done = calyx.cell "c0" @A : i1, i1, i1, i1, i1
+    %c1_1 = constant 1 : i1
+    calyx.wires { calyx.group @Group1 { calyx.group_done %c1_1 : i1 } }
+    calyx.control {
+      calyx.seq {
+        // expected-error @+1 {{empty 'then' region.}}
+        calyx.if %c0.out, @Group1 {} else {}
+      }
+    }
+  }
+}
+
+// -----
+
+calyx.program {
+  calyx.component @A(%go: i1, %clk: i1, %reset: i1) -> (%out: i1, %done: i1) {
+    calyx.wires {}
+    calyx.control {}
+  }
+  calyx.component @main(%in: i32, %go: i1, %clk: i1, %reset: i1) -> (%out: i32, %done: i1) {
+    %c0.go, %c0.clk, %c0.reset, %c0.out, %c0.done = calyx.cell "c0" @A : i1, i1, i1, i1, i1
+    calyx.wires { }
+    calyx.control {
+      calyx.seq {
+        // expected-error @+1 {{expected ','}}
```

Comment:
I don't think this test is necessary, since you're essentially testing ODS's assemblyFormat as this point. We should leave tests for the `VerifyIfOp`.

Reply:
Agreed; i was in doubt about this test given what you mentioned, but defaulted to leave it.